cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -0,0 +1,692 @@
+
+		})
+
+		It("Update an existing Application table in the database, call Reconcile on the Argo CD Application, and verify an existing ApplicationState DB entry is updated", func() {
+			// Close database connection.
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			ctx = context.Background()
+
+			// Add databaseID label to applicationID.
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			applicationState := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+				Health:                          "Progressing",
+				Sync_Status:                     "OutOfSync",
+			}
+
+			// Create Application in Database.
+			err = reconciler.DB.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+
+			// Create Application in Database.
+			err = reconciler.DB.CreateApplicationState(ctx, applicationState)
+			Expect(err).To(BeNil())
+
+			// Create a new ArgoCD Application.
+			err = reconciler.Create(ctx, guestbookApp)
+			Expect(err).To(BeNil())
+
+			// Call reconcile function.
+			result, err := reconciler.Reconcile(ctx, newRequest(namespace, name))
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+
+			// Verify Application State is created in database.
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationState)
+			Expect(err).To(BeNil())
+
+			// Verify that Health and Status of ArgoCD Application is equal to the Health and Status of Application in database.
+			Expect(applicationState.Health).To(Equal(string(guestbookApp.Status.Health.Status)))
+			Expect(applicationState.Sync_Status).To(Equal(string(guestbookApp.Status.Sync.Status)))
+
+		})
+
+		It("Application CR missing corresponding namespace entry", func() {
+			// Close database connection.
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			ctx = context.Background()
+
+			// Add databaseID label to applicationID.
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			// Create Application in Database.
+			err = reconciler.DB.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+
+			// Call reconcile function.
+			result, err := reconciler.Reconcile(ctx, newRequest(namespace, name))
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+
+		})
+
+		It("Application CR missing corresponding database entry", func() {
```

Comment:
Can likewise change this to something like "has an Argo CD Application that points to a database entry that doesn't exist"